src/RP2040.Core/Cpu/Instructions/BitOps.cs

```diff
@@ -146,6 +146,19 @@ public static void MovRegister(ushort opcode, CortexM0Plus cpu)
 		
 		ptrRd = valRm;
 	}
+
+	[MethodImpl(MethodImplOptions.AggressiveInlining)]
+	public static void MovsImm8(ushort opcode, CortexM0Plus cpu)
+	{
+		var value = (uint)(opcode & 0xFF);
+		var rd = (opcode >> 8) & 7;
+
+		ref var ptrRd = ref cpu.Registers[rd];
```

Comment:
This line can be reduced to a single assignment, since the references are obtained only when we will be writing back and reuse the value multiple times, but for this, I think it should be fine to have only a direct assigment